<?xml version="1.0"?>
<launch>
    <node pkg="amcl" type="amcl" name="amcl" output="screen">
        <param name="base_frame_id" value="base_footprint"/>
        <param name="use_map_topic" value="true"/>
        <param name="first_map_only" value="false"/>
        <!-- -->

        <!-- Publish scans from best pose at a max of 10 Hz -->
        <param name="odom_model_type" value="diff"/>
        <param name="gui_publish_rate" value="10.0"/>

        <param name="laser_max_beams" value="180"/>
        <param name="laser_min_range" value="0.0"/>
        <param name="laser_max_range" value="6.0"/>
        <param name="laser_z_hit" value="0.5"/>
        <param name="laser_z_rand" value="0.5"/>
        <param name="laser_z_short" value="0.05"/>
        <param name="laser_sigma_hit" value="0.2"/>
        <param name="laser_lambda_short" value="0.1"/>
        <param name="laser_z_max" value="0.05"/>
        <param name="laser_model_type" value="likelihood_field"/>
        <param name="laser_likelihood_max_dist" value="2.0"/>

        <param name="min_particles" value="500"/>
        <param name="max_particles" value="2000"/>

        <param name="kld_err" value="0.05"/>
        <param name="kld_z" value="0.99"/>

        <param name="odom_alpha1" value="2.0"/>
        <param name="odom_alpha2" value="1.0"/>
        <!-- translation std dev, m -->
        <param name="odom_alpha3" value="0.5"/>
        <param name="odom_alpha4" value="0.5"/>
        <param name="odom_alpha5" value="0.1"/>

        <param name="update_min_d" value="0.05"/>
        <param name="update_min_a" value="0.05"/>

        <param name="odom_frame_id" value="odom"/>
        <param name="resample_interval" value="1"/>

        <param name="transform_tolerance" value="1.0"/>
        <param name="recovery_alpha_slow" value="0.0"/>
        <param name="recovery_alpha_fast" value="0.0"/>

        <param name="initial_pose_x" value="0.0"/>
        <param name="initial_pose_y" value="0.0"/>
        <param name="initial_pose_a" value="0.0"/>

        <param name="scan" value="scan"/>
    </node>
</launch>
